### ISSUE TITLE
[BUG] Possible Dependency Peering Issue

### ISSUE THREAD

**Issue by saengowp** [thinc-org/cugetreg#489]
**Describe the bug**

During the investigation of the endpoint `/auth/refreshtoken`, in which the method correctly throwed `BadRequestException`, but the resulting HTTP code is 5xx instead of the expected 400.

I bisected the problem to the commit c4227b. The commit added a new NestJS subproject to the monorepo. I narrowed it further and found that removing the `@nestjs/platform-express` dependency in `admin-api` make the issue disappear. However, I do not know turbo and pnpm enough to understand why removing Dependency from one subproject caused changes in another subproject.

 I guessed that the cause is likely from `@nest/core` and `@nest/common` having two version of `HTTPException`, this have not been definitively confirmed however.

**To Reproduce**
1. Go to the offending commit
2. Send curl POST empty body to `/_api/auth/refreshtoken`

**Expected behavior**

Expect 400 no auth cookie

Actually get 500.

**Comment by bombnp** [comment]
Where exactly does it throw 500?

**Comment by bombnp** [comment]
![image](https://user-images.githubusercontent.com/24814968/216807239-04ad30b5-0951-4f77-ad23-335eb3979704.png)
oh dear, another case of having multiple entries of having the same version of a dependency. A week ago, I was fixing a bug related to reg scraper and it turned out it has multiple entries of `@nestjs/core` which caused it to resolve incorrectly. This may be related.

**Comment by bombnp** [comment]
`admin-api` and `reg-scraper` seems to be depending on `9.2.1_mnr6j2del53muneqly5h4y27ai`, while `api` is depending on `9.2.1_huyc3vjxdj723rrinn6lunzxyq`
![image](https://user-images.githubusercontent.com/24814968/216807393-cb3eb52f-7a33-4319-be91-6c3fe8a0e71a.png)
![image](https://user-images.githubusercontent.com/24814968/216807404-9ac89834-2771-4b33-82ee-bef5b3c84522.png)

**Comment by bombnp** [comment]
I've determined the cause. @saengowp 

Both `admin-api` and `api` uses the same entry of `@nestjs/platform-express` (`9.2.1_hjcqpoaebdr7gdo5hgc22hthbe`), but has different entries for `@nestjs/common` (admin-api: `9.2.1_mnr6j2del53muneqly5h4y27ai`, api: `9.2.1_huyc3vjxdj723rrinn6lunzxyq`).

However, `@nestjs/platform-express:9.2.1_hjcqpoaebdr7gdo5hgc22hthbe` has `9.2.1_mnr6j2del53muneqly5h4y27ai` as its dependency, which does not match the one in `api`, which might have caused the nest platform to think that api's `HTTPException` is not the same as the platform's `HTTPException` (since it uses different `@nestjs/common`).

The reason there's two entries of `@nestjs/common`, is because api has `class-validator` and `class-transformer` as its dependencies, which are optional peer dependencies of `@nestjs/common`. According to [pnpm's issue](https://github.com/pnpm/pnpm/issues/5585), when two projects have the exact same dependency name and version, if it has different peer dependencies, IT WILL STILL CREATE MULTIPLE COPIES OF THE EXACT SAME VERSION.
![image](https://user-images.githubusercontent.com/24814968/216808459-93cd90ab-422e-4301-a558-f9f16c20552a.png)

**Comment by bombnp** [comment]
TL;DR install class-validator and class-transformer as dependencies in all nestjs apps, or switch to yarn lol (@saenyakorn suggested this)

**Comment by saengowp** [comment]
Thank you for a very thorough investigation. This feels like having a
loaded gun pointed at our feet, ready to go whenever we modify the
dependencies. 🥲

Nut Pinyo ***@***.***> schrieb am So., 5. Feb. 2023, 15:23:

**Comment by bombnp** [comment]
Should i merge the fix in for now?

**Comment by saengowp** [comment]
Sure!

Nut Pinyo ***@***.***> schrieb am So., 5. Feb. 2023, 16:39:
